CustomTextAnalytics.CognitiveSearch/Solution/IndexingCliTool/CognitiveSearchIntegration.Core/Services/Indexers/CustomTextIndexingService.cs

```diff
@@ -62,6 +62,12 @@ public SearchIndex CreateIndex(CustomTextSchema schema, string indexName)
                 IsKey = true
             });
 
+            // document name
+            indexFields.Add(new SearchField("document_name", SearchFieldDataType.String));
```

Comment:
did you read the docs?
i.e. will document name in the index cause any problems
like service limitations or something?

Reply:
Yes I did read the docs. the existing limitation I found happens only in `document key` where you can't add unsafe url characters and it can't be longer than 1024 characters. check here: https://docs.microsoft.com/en-us/azure/search/search-indexer-field-mappings#field-mapping-functions

Please if you know something else tell me.